src/Orleans/Runtime/AsynchQueueAgent.cs

```diff
@@ -30,7 +30,10 @@ public void QueueRequest(T request)
                 queueTracking.OnEnQueueRequest(1, requestQueue.Count, request);
             }
 #endif
-            requestQueue.Add(request);
+            if (requestQueue != null)
+            {
+                requestQueue.Add(request);
```

Comment:
I think it's better due to potential race conditions too :)
But yeah, even better now that it doesn't even attempt to touch statistics.